#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_fix_new_theora.dpatch by  <root@localhost>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix recording with newest libtheora versions

@DPATCH@

--- recordmydesktop-0.3.8.1+svn602.orig/src/rmd_initialize_data.c	2010-02-08 18:30:42.000000000 +0100
+++ recordmydesktop-0.3.8.1+svn602.bueno/src/rmd_initialize_data.c	2010-02-08 19:12:36.000000000 +0100
@@ -223,7 +223,7 @@
     args->channels             = 1;
     args->frequency            = 22050;
     args->buffsize             = 4096;
-    args->v_bitrate            = 45000;
+    args->v_bitrate            = 0;
     args->v_quality            = 63;
     args->s_quality            = 10;
 
@DPATCH@

--- recordmydesktop-0.3.8.1+svn602.orig/src/rmd_parseargs.c	2010-02-08 18:30:42.000000000 +0100
+++ recordmydesktop-0.3.8.1+svn602.bueno/src/rmd_parseargs.c	2010-02-08 19:15:01.000000000 +0100
@@ -215,7 +215,7 @@
 
         { "v_bitrate", '\0',
           POPT_ARG_INT, &arg_return->v_bitrate, 0,
-          "A number from 45000 to 2000000 for desired encoded video bitrate(default 45000).",
+          "A number from 0 to 2000000 for desired encoded video bitrate(default 0).",
           "n" },
 
         { "s_quality", '\0',
@@ -460,8 +460,8 @@
         fprintf(stdout, "--v_quality must be within the inclusive range [0-63].\n");
         success = FALSE;
     }
-    if (args->v_bitrate < 45000 || args->v_quality > 2000000) {
-        fprintf(stdout, "--v_bitrate must be within the inclusive range [45000-2000000].\n");
+    if (args->v_bitrate < 0 || args->v_quality > 2000000) {
+        fprintf(stdout, "--v_bitrate must be within the inclusive range [0-2000000].\n");
         success = FALSE;
     }
     if (args->frequency <= 0) {
